cli/commands/project_sync.go

```diff
@@ -0,0 +1,235 @@
+package commands
+
+import (
+  "fmt"
+  "io/ioutil"
+  "net"
+  "os"
+  "os/exec"
+  "path"
+  "strings"
+  "time"
+
+  "github.com/phase2/rig/cli/commands/project"
+  "github.com/urfave/cli"
+  "gopkg.in/yaml.v2"
+)
+
+type ProjectSync struct {
+  BaseCommand
+}
+
+// Minimal compose file struct to discover volumes
+type ComposeFile struct {
+  Volumes map[string]Volume
+}
+
+type Volume struct {
+  External bool
+}
+
+const UNISON_PORT=5000
+const MAX_WATCHES="100000"
+
+func (cmd *ProjectSync) Commands() []cli.Command {
+  start := cli.Command{
+    Name:        "sync:start",
+    Usage:       "Start a unison sync on local project directory. Optionally provide a volume name.",
+    ArgsUsage:   "[optional volume name]",
+    Description: "Volume name will be discovered in the following order: argument to this command > outrigger project config > docker-compose file > current directory name",
+    Before:      cmd.Before,
+    Action:      cmd.RunStart,
+  }
+  stop := cli.Command{
+    Name:        "sync:stop",
+    Usage:       "Stops a unison sync on local project directory. Optionally provide a volume name.",
+    ArgsUsage:   "[optional volume name]",
+    Description: "Volume name will be discovered in the following order: argument to this command > outrigger project config > docker-compose file > current directory name",
+    Before:      cmd.Before,
+    Action:      cmd.RunStop,
+  }
+
+  return []cli.Command{start, stop}
+}
+
+// Start the unison sync process
+func (cmd *ProjectSync) RunStart(ctx *cli.Context) error {
+  project.ConfigInit()
+  config := project.GetProjectConfigFromFile(project.GetConfigPath())
+  volumeName := cmd.GetVolumeName(ctx, config)
+  cmd.out.Verbose.Printf("Starting sync with volume: %s", volumeName)
+
+  // Ensure the processes can handle a large number of watches
+  if err := cmd.machine.SetSysctl("fs.inotify.max_user_watches", MAX_WATCHES); err != nil {
+    cmd.out.Error.Fatalf("Error configuring watches on Docker Machine: %v", err)
+  }
+
+  cmd.out.Info.Printf("Starting sync volume: %s", volumeName)
+  exec.Command("docker","volume", "create", volumeName).Run()
+
+  cmd.out.Info.Println("Starting unison container")
+  exec.Command("docker","container", "stop", volumeName).Run()
+  err := exec.Command("docker","container", "run", "--detach", "--rm",
+    "-v", fmt.Sprintf("%s:/unison", volumeName),
+    "-e", "UNISON_DIR=/unison",
+    "-l", fmt.Sprintf("com.dnsdock.name=%s", volumeName),
+    "-l", "com.dnsdock.image=volume.outrigger",
```

Comment:
Yeah it is. Turns out we dont ever use that domain name though, so really it can just go away. I inspect the containers and use the IP address due to that cgo issue we rolled back